src/main/java/org/jenkinsci/plugins/oic/OicAlgorithmValidator.java

```diff
@@ -0,0 +1,145 @@
+package org.jenkinsci.plugins.oic;
+
+import com.nimbusds.jose.Algorithm;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.impl.AESCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ContentCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ECDHCryptoProvider;
+import com.nimbusds.jose.crypto.impl.PasswordBasedCryptoProvider;
+import com.nimbusds.jose.crypto.impl.RSACryptoProvider;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import jenkins.security.FIPS140;
+
+/**
+ * This class helps in validating algorithms for FIPS compliance and filtering the non-compliant algorithms when in
+ * FIPS mode.
+ */
+public class OicAlgorithmValidator {
+
+    private static final boolean isFIPSMode = FIPS140.useCompliantAlgorithms();
+    private static final Set<JWSAlgorithm> jwsSupportedAlgorithms = new LinkedHashSet<>();
+    private static final Set<JWEAlgorithm> jweSupportedAlgorithms = new LinkedHashSet<>();
+    private static final Set<EncryptionMethod> supportedEncryptionMethod = new LinkedHashSet<>();
+
+    // Below list of compliant algorithms will be used to block the FIPS non-compliant algorithms.
+    static {
+        // Init compliant Jws algorithms
+        jwsSupportedAlgorithms.addAll(MACSigner.SUPPORTED_ALGORITHMS);
+        jwsSupportedAlgorithms.addAll(RSASSASigner.SUPPORTED_ALGORITHMS);
+        jwsSupportedAlgorithms.addAll(ECDSASigner.SUPPORTED_ALGORITHMS);
+
+        // Init compliant Jwe algorithms
+        jweSupportedAlgorithms.addAll(AESCryptoProvider.SUPPORTED_ALGORITHMS);
+        jweSupportedAlgorithms.addAll(RSACryptoProvider.SUPPORTED_ALGORITHMS);
+        // RSA1_5 is deprecated and not a compliant algorithm.
+        jweSupportedAlgorithms.remove(JWEAlgorithm.RSA1_5);
+        jweSupportedAlgorithms.addAll(ECDHCryptoProvider.SUPPORTED_ALGORITHMS);
+        jweSupportedAlgorithms.addAll(PasswordBasedCryptoProvider.SUPPORTED_ALGORITHMS);
+
+        // Init complaint EncryptionMethods
+        supportedEncryptionMethod.addAll(ContentCryptoProvider.SUPPORTED_ENCRYPTION_METHODS);
+    }
+
+    /**
+     * Checks if the Jws signing algorithm used for OIC configuration is FIPS compliant.
+     */
+    public static boolean isJwsAlgorithmFipsNonCompliant(Algorithm algorithm) {
```

Comment:
As a rule of thumb, "positive" method (e.g.: `isJwsAlgorithmCompliant`) are easier to follow (those double negations make this difficult to read)

Reply:
Done